izpack-compiler/src/main/java/com/izforge/izpack/compiler/packager/impl/PackagerBase.java

```diff
@@ -381,6 +384,41 @@ protected boolean packSeparateJars()
         return info != null && info.getWebDirURL() != null;
     }
 
+    private List<DynamicVariable> buildVariableList()
+    {
+        DirectedGraph<DynamicVariable> graph = new DirectedGraph<DynamicVariable>();
+        DynamicVariable rootVar = new DynamicVariableImpl("", "");
+
+        for (List<DynamicVariable> dynVariables : dynamicVariables.values())
+        {
+            for (DynamicVariable var : dynVariables)
+            {
+                graph.addEdge(rootVar, var);
+                for (String childName : var.getUnresolvedVariableNames())
+                {
+                    List<DynamicVariable> childVars = dynamicVariables.get(childName);
+                    if (childVars != null)
+                    {
+                        for (DynamicVariable childVar : childVars)
+                        {
+                            graph.addEdge(var, childVar);
+                        }
+                    }
+                }
+            }
+        }
+
+        Iterator<DynamicVariable> it = new BreadthFirstIterator<DynamicVariable>(graph, rootVar);
+        List<DynamicVariable> nodes = new ArrayList<DynamicVariable>();
+        while (it.hasNext()) {
+            nodes.add(it.next());
+        }
+        nodes.remove(rootVar);
```

Comment:
Hi René,
 
If you find something useful for this problem let me know, please.
 
Yes, I had two ideas. The first one is already proven false :-(
 
Regards
 Tom



---
Mail & Cloud Made in Germany mit 3 GB Speicher! Jetzt kostenlos anmelden